n2y/notion.py

```diff
@@ -46,6 +47,35 @@
 # TODO: Rename this file `client.py`
 
 
+def retry_api_call(api_call):
```

Comment:
I like that you're using a decorator. Nice idea!

I know there are a few popular python libraries that provide a retry decorator ([retry](https://github.com/invl/retry), [tenacity](https://github.com/jd/tenacity) are two that I know of). I think we should use one of them instead of implementing our own. You can probably swap their decorator for yours as-is. It'll also make it easier for us to add more advanced retry functionality like backoffs, jitter, choose which exceptions to retry on, etc later on. What do you think @sHermanGriffiths and @johndgiese?

Also @sHermanGriffiths, if we do implement our own decorator, we need a good unit test to make sure it works.

Reply:
I like the idea of using a library for this.

Reply:
I like the idea of using a library as well, but I can't find any that parse the delay time from the exception itself, which is the premise of the functionality we're trying to create here.

Reply:
@sHermanGriffiths that's a great point, I think we should use the header sent back from Notion to figure out the timeout time.

Reply:
@sHermanGriffiths Ah I see. Great point and that makes total sense!

I'll leave the decision up to you. If you do move forward with a custom implementation though, please include a test for it.